blocks/search/search.js

```diff
@@ -104,172 +100,128 @@ function addSearchByListeners(wrapper, form) {
+  const searchType = urlParams.get('st');
+  const category = urlParams.get('category');
+  const targetOffset = isFirstSet && offsetParam === '0' ? 0 : parseInt(offsetParam) + 1;
+  if (!isFirstSet) {
+    const newUrl = new URL(window.location);
+    newUrl.searchParams.set('offset', targetOffset);
+    window.history.pushState({ path: newUrl.href }, '', newUrl.href);
+  }
+  const filtersWrapper = document.querySelector('.filters-wrapper');
+  const loadingLabel = getTextLabel('loading_label');
+  let loadingElement = document.querySelector('.loading');
+  if (!loadingElement) {
+    loadingElement = createElement('div', { classes: 'loading' });
+    resultsSection.append(loadingElement);
+  }
+  loadingElement.textContent = loadingLabel;
+  const offset = targetOffset * parseInt(MAX_PRODUCTS_PER_QUERY);
+  const searchParams = { query, offset, make, model, searchType, category };
+  const { results, categories } = await fetchSearchResults(searchParams);
+  loadingElement.remove();
+  const searchResultsSection = document.querySelector('.search-results-section');
+  const titleElement = searchResultsSection.querySelector('.title');
+  if (results?.length > 0) {
+    results.forEach((result) => {
+      const liElement = productCard(result, searchType);
+      resultsList.appendChild(liElement);
+    });
+    const titleContent = getTextLabel('search_results_title');
+    const type = searchType === 'cross' ? 'cross-reference' : 'parts';
+    const titleText = `${titleContent} ${searchType === 'cross' ? `${type}: "${query}"` : `${make || ''} ${model || ''} ${query} ${type}`}`;
+    titleElement.textContent = titleText;
+    const buttonTextContent = getTextLabel('pagination_button');
+    const resultsCountElement = document.querySelector('.displayed-text');
+    const currentAmount = document.querySelectorAll('.product-card').length;
+    const displayedTextContent = getTextLabel('pagination_text');
+    const newText = displayedTextContent.replace('[$]', currentAmount);
+    resultsCountElement.innerText = newText;
+    if (targetOffset === 0) {
+      const bottomMoreBtn = createElement('button', { classes: ['more-button', 'bottom-more-button'] });
+      bottomMoreBtn.textContent = buttonTextContent;
+      resultsSection.appendChild(bottomMoreBtn);
+      bottomMoreBtn.onclick = () => getAndApplySearchResults({ isFirstSet: false });
+    }
+    if (results.length < parseInt(MAX_PRODUCTS_PER_QUERY)) {
+      document.querySelectorAll('.more-button').forEach((moreBtn) => moreBtn.remove());
+    }
+  }
+  if (!results || results.length === 0) {
+    const titleText = getTextLabel('no_results_title').replace('[$]', `${query}`);
+    titleElement.innerText = titleText;
+    const fragment = document.createRange().createContextualFragment(noResultsTemplate);
+    searchResultsSection.classList.add('no-results');
+    searchResultsSection.insertBefore(fragment, filtersWrapper);
+  }
+  const filters = buildFilter(categories);
+  if (filters) {
```

Comment:
`filtersWrapper ` may not exist in the dom. Please add a check for it on this `if`

Reply:
Done